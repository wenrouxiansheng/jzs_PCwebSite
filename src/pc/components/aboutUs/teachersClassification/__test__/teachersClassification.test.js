import React from "react";
import renderer from "react-test-renderer";
import TeachersClassification from "../index";
import { shallow } from "enzyme";
import { BrowserRouter as Router } from "react-router-dom";

describe("PC:<TeachersClassification />", () => {
  it("render", () => {
    const wrapper = renderer.create(
      <Router>
        <TeachersClassification />
      </Router>
    );
    let tree = wrapper.toJSON();
    //渲染测试
    expect(tree).toMatchSnapshot();
  });

  const wrapper = shallow(<TeachersClassification />);
  it("paging event", () => {
    const NUM = 3;

    wrapper.find("Pagination").simulate("change", NUM);
    expect(wrapper.find("Pagination").prop("current")).toBe(NUM);
  });

  it("teacher class event", () => {
    const TYPE = 1;
    wrapper.find(".nav li").at(TYPE).simulate("click");
    expect(wrapper.state("category")).toBe(TYPE);
  });
});
