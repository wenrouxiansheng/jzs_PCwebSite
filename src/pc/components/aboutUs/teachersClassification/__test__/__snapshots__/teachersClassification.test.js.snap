// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PC:<TeachersClassification /> render 1`] = `
<div
  className="teachersClassification"
  style={Object {}}
>
  <div
    className="animationTitle"
    style={Object {}}
  >
    <h3>
      <b>
        师资力量
      </b>
    </h3>
    <h6>
      汇聚国内外艺术院校师资力量
    </h6>
  </div>
  <ul
    className="nav"
  >
    <li
      className="active"
      onClick={[Function]}
    >
      全部
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      舞蹈
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      声乐
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      器乐
    </li>
    <li
      className=""
      onClick={[Function]}
    >
      美术
    </li>
  </ul>
  <div
    className="teacherBox"
  >
    <div
      className="teacher"
    >
      <a
        href="/site/pc/aboutUs/teachers/detail/1"
        onClick={[Function]}
      >
        <img
          alt=""
        />
        <h3>
          许杨
        </h3>
        <p>
          <em>
            教龄
          </em>
          ：
          <span>
            9年
          </span>
        </p>
        <p
          className="text"
        >
          <em>
            教学课程
          </em>
          ：
          <span>
            童声、美声、民族、通俗唱法
          </span>
        </p>
      </a>
    </div>
    <div
      className="teacher"
    >
      <a
        href="/site/pc/aboutUs/teachers/detail/2"
        onClick={[Function]}
      >
        <img
          alt=""
        />
        <h3>
          孙亚秋
        </h3>
        <p>
          <em>
            教龄
          </em>
          ：
          <span>
            4年
          </span>
        </p>
        <p
          className="text"
        >
          <em>
            教学课程
          </em>
          ：
          <span>
            儿童画，创意美术，水彩，水粉，素描，彩铅，油画
          </span>
        </p>
      </a>
    </div>
    <div
      className="teacher"
    >
      <a
        href="/site/pc/aboutUs/teachers/detail/3"
        onClick={[Function]}
      >
        <img
          alt=""
        />
        <h3>
          孟繁颖
        </h3>
        <p>
          <em>
            教龄
          </em>
          ：
          <span>
            7年
          </span>
        </p>
        <p
          className="text"
        >
          <em>
            教学课程
          </em>
          ：
          <span>
            古典舞、芭蕾舞、民族民间舞、爵士舞、踢踏舞、肚皮舞
          </span>
        </p>
      </a>
    </div>
    <div
      className="teacher"
    >
      <a
        href="/site/pc/aboutUs/teachers/detail/4"
        onClick={[Function]}
      >
        <img
          alt=""
        />
        <h3>
          任雅臣
        </h3>
        <p>
          <em>
            教龄
          </em>
          ：
          <span>
            5年
          </span>
        </p>
        <p
          className="text"
        >
          <em>
            教学课程
          </em>
          ：
          <span>
            流行唱法、美声唱法、民族唱法、音基
          </span>
        </p>
      </a>
    </div>
    <div
      className="teacher"
    >
      <a
        href="/site/pc/aboutUs/teachers/detail/5"
        onClick={[Function]}
      >
        <img
          alt=""
        />
        <h3>
          向定军
        </h3>
        <p>
          <em>
            教龄
          </em>
          ：
          <span>
            5年
          </span>
        </p>
        <p
          className="text"
        >
          <em>
            教学课程
          </em>
          ：
          <span>
            架子鼓、架子鼓考级、少儿架子鼓
          </span>
        </p>
      </a>
    </div>
    <div
      className="teacher"
    >
      <a
        href="/site/pc/aboutUs/teachers/detail/6"
        onClick={[Function]}
      >
        <img
          alt=""
        />
        <h3>
          李东伟
        </h3>
        <p>
          <em>
            教龄
          </em>
          ：
          <span>
            4年
          </span>
        </p>
        <p
          className="text"
        >
          <em>
            教学课程
          </em>
          ：
          <span>
            古典吉他，民谣吉他，电吉他，尤克里里
          </span>
        </p>
      </a>
    </div>
    <div
      className="teacher"
    >
      <a
        href="/site/pc/aboutUs/teachers/detail/7"
        onClick={[Function]}
      >
        <img
          alt=""
        />
        <h3>
          张培佩
        </h3>
        <p>
          <em>
            教龄
          </em>
          ：
          <span>
            3年
          </span>
        </p>
        <p
          className="text"
        >
          <em>
            教学课程
          </em>
          ：
          <span>
            芭蕾舞、古典舞、民族民间舞、爵士舞、少儿舞蹈
          </span>
        </p>
      </a>
    </div>
  </div>
  <div
    className="pagination"
  >
    <ul
      className="ant-pagination"
      style={Object {}}
      unselectable="unselectable"
    >
      <li
        aria-disabled={true}
        className="ant-pagination-prev ant-pagination-disabled"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={null}
        title="Previous Page"
      >
        <button
          className="ant-pagination-item-link"
          disabled={true}
          tabIndex={-1}
          type="button"
        >
          <span
            aria-label="left"
            className="anticon anticon-left"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </span>
        </button>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={1}
      >
        <a
          rel="nofollow"
        >
          1
        </a>
      </li>
      <li
        aria-disabled={true}
        className="ant-pagination-next ant-pagination-disabled"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={null}
        title="Next Page"
      >
        <button
          className="ant-pagination-item-link"
          disabled={true}
          tabIndex={-1}
          type="button"
        >
          <span
            aria-label="right"
            className="anticon anticon-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </button>
      </li>
    </ul>
  </div>
</div>
`;
